### PR TITLE
Reorder Crons with drag and drop

### PR DESCRIPTION
**Pull request by Sorbitol**
Adds drag and drop functionality to reorder Crons via long press.

- Cron position is saved in the new database column "position".
- Added database migration and tests for migrations.
- Added callback class for ItemTouchHelper along with interface
  DragInterface which NotificationCronAdapter implements.
- Added method to update database after Cron move to ButtonListener
  interface and renamed it ViewListener since it no longer just deals
  with the button.
- Implemented methods in main Activity, ViewModel and Dao to update
  database after Cron move.

### CODE REVIEW

File: app/src/main/java/com/github/fi3te/notificationcron/data/local/AppDatabase.kt [hunk 1]
```diff
@@ -16,12 +19,19 @@ abstract class AppDatabase : RoomDatabase() {
...
+        val MIGRATION_1_2: Migration = object : Migration(1, 2) {
             override fun migrate(database: SupportSQLiteDatabase) {
                 database.execSQL("ALTER TABLE notification_cron ADD COLUMN enabled INTEGER DEFAULT 1 NOT NULL")
             }
         }
 
+        val MIGRATION_2_3: Migration = object : Migration(2, 3) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                database.execSQL("ALTER TABLE notification_cron ADD COLUMN position INTEGER")
+                database.execSQL("UPDATE notification_cron SET position = id WHERE position IS NULL;")
```
Comment (fi3te) [inline app/src/main/java/com/github/fi3te/notificationcron/data/local/AppDatabase.kt:31]: Please unify the use of semicolons in database.execSQL calls.

**Comment by fi3te** [comment]
Hi Sorbitol,

many thanks for your contribution! 

The changes look good. I have added one small remark.

Best regards,
Fiete

File: app/src/main/java/com/github/fi3te/notificationcron/data/local/AppDatabase.kt (hunk 1 above)
Comment (Sorbitol) [reply app/src/main/java/com/github/fi3te/notificationcron/data/local/AppDatabase.kt]: Done!

**Comment by Sorbitol** [comment]
Hi Fiete,

Thank you for reviewing my PR. I have added a commit to remove the redundant semicolon.

Best regards,
Sorbitol

**Pull request merged**